### PR TITLE
Rename Vision Processing Unit(VPU) to Neural Processing Unit. (NPU)

### PR DESCRIPTION
**Pull request by saurabhkale17**
### Description
This PR is for renaming VPU device to NPU.
This PR also includes addition of ops support for NPU and fixes for some of the C++ and python unit tests for NPU.

### Motivation and Context
https://jira.devtools.intel.com/browse/HAFP-2430

### CODE REVIEW

File: docs/python/ReadMeOV.rst
Comment (sfatimar) [inline docs/python/ReadMeOV.rst:10]: Let us remove from ReadME until we have approval from CCG

**Comment by sfatimar** [comment]
Please remove from ReadMeOV.rst so we can merge

File: docs/python/ReadMeOV.rst
Comment (saurabhkale17) [reply docs/python/ReadMeOV.rst]: Removed NPU from ReadME @sfatimar

**Review by sspintel — APPROVED** [review]
LGTM

**Pull request merged**
